doc/nrf/ug_bootloader_external_flash.rst

```diff
@@ -0,0 +1,66 @@
+.. _ug_bootloader_external_flash:
+
+Using external flash memory partitions
+######################################
+
+It is possible to use the external flash memory as the storage partition for the secondary slot.
+This requires a driver for the external flash memory that supports the following:
+
+* Single-byte read and write.
+* Writing data from the internal flash memory to the external flash memory.
+
+See :ref:`pm_external_flash` for general information about how to set up partitions in the external flash memory in the Partition Manager.
+
+.. note::
+
+   External flash memory partitions for image slots are currently only supported by MCUboot.
+
+To configure a bootloader to use the external flash memory for the secondary slot, edit the configuration file of the bootloader child image to specify an external flash region.
+For MCUboot, this configuration is the file :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml`.
+Define the region under ``mcuboot_secondary``:
+
+.. code-block:: yaml
+
+   mcuboot_secondary:
+       region: external_flash
+       size: CONFIG_PM_EXTERNAL_FLASH_SIZE
+
+The nRF52840 DK comes with the external flash memory that can be used for the secondary slot and be accessed using the QSPI NOR flash memory driver.
+The following steps demonstrate how to configure an application for the nRF52840 DK:
+
+1. Add the following configuration options to the ``prj.conf`` file associated to the MCUboot child image:
+
+   .. code-block:: console
+
+      CONFIG_NORDIC_QSPI_NOR=y
+      CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
+      CONFIG_MULTITHREADING=y
+      CONFIG_BOOT_MAX_IMG_SECTORS=256
+      CONFIG_PM_EXTERNAL_FLASH=y
+      CONFIG_PM_EXTERNAL_FLASH_DEV_NAME="MX25R64"
+      CONFIG_PM_EXTERNAL_FLASH_SIZE=0xf4000
```

Comment:
@sigvartmh perhaps you can help clarify this?

Reply:
This is an issue in partition manager where information about the size cannot be shared into something placed into a region(which is what the external flash is defined as).
So currently the `mcuboot_secondary` size must be defined as the same size as the `mcuboot_primary`. I do agree this should be documented better in some way.

Also it should be made easier for a customer to enable.

Reply:
@kkasperczyk-no If there is a joint opinion that this should be clarified in the docs, is this something we can raise in a separate issue/PR and edit there? Or is this blocking users following the current docs here?

Reply:
@kyberdin I don't have strong opinion on that, I'm just saying that I followed the instruction and for me it failed with some error that description doesn't clearly suggest what actions to take in order to solve it. If it failed for me, there is a chance that it will fail for anybody else (actually I know at least one Nordic's customer that had problem with that) and it's not ok to leave someone alone with this problem.

If guys more familiar with MCUboot and partition manager claim, that it is something bigger, maybe we just should inform user that he/she may face with such issue and how to solve it. I would recommend adding here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html#configuring-static-partitions a note after the following sentence:

"The current partition configuration for a build can be found in $BUILD_DIR/partitions.yml. To apply the current configuration as a static configuration, copy this file to $APPLICATION_SOURCE_DIR/pm_static.yml."

.. note::
   Please verify in the pm_static.yml that generated `mcuboot_primary` and `mcuboot_secondary` sizes are the same, as it is required by the MCUboot. In case the values differ, align manually the `mcuboot_secondary` size value to be equal to the `mcuboot_primary` size.

Moreover I would add additional information below the configuration section that we are discussing. After the sentence: "These options enable the QSPI NOR flash memory driver and the external flash memory of the nRF52840 DK."

.. note::
   Set `CONFIG_PM_EXTERNAL_FLASH_SIZE` value has direct impact on the size of `mcuboot_secondary` partition. That value has to match the `mcuboot_primary` partition size and not meeting this requirement will result in seeing `Cannot upgrade: not a compatible amount of sectors` error in the device logs. You can check the actual partition sizes after building the target by invoking in the build directory following command:
$ ninja partition_manager_report

Those are just my suggestions, so please consider rephrasing/changing content as it seems appropriate.

Reply:
@kkasperczyk-no I have created NCSDK-9847 to update the documentaiton for your suggestions.

Reply:
Thanks, in case we are tracking it in Jira, I don't want to block merging it. Removing request changes mark.